GitCommands/Git/GitGpgController.cs

```diff
@@ -0,0 +1,219 @@
+        /// </summary>
+        /// <returns>Enum value that indicate if current git revision has one tag with good signature, one tag with bad signature or more than one tag.</returns>
+        TagStatus CheckTagSign();
+
+        /// <summary>
+        /// Obtain the number of tag on current git revision.
+        /// </summary>
+        /// <returns>Returns the number of annotated or signed tag on current revision.</returns>
+        int NumberOfTag { get; }
+
+        /// <summary>
+        /// Obtain the tag verification message for all the tag in current git revision 
+        /// </summary>
+        /// <returns>Full concatenated string coming from GPG analysis on all tags on current git revision.</returns>
+        string TagVerifyMessage { get; }
+    }
+
+
+    public class GitGpgController : IGitGpgController
+    {
+        private IGitModule _module;
+        private GitRevision _revision;
+
+        /* Commit GPG status */
+        private const string GOOD_SIGN = "G";
+        private const string BAD_SIGN = "B";
+        private const string UNK_SIGN_VALIDITY = "U";
+        private const string EXPIRED_SIGN = "X";
+        private const string EXPIRED_SIGN_KEY = "Y";
+        private const string REVOKED_KEY = "R";
+        private const string MISSING_PUB_KEY = "E";
+        private const string NO_SIGN = "N";
+
+
+        /* Tag GPG status */
+        private const string _goodSignature = "GOODSIG";
+        private const string _validTagSign = "VALIDSIG";
+        private const string _noTagPubKey = "NO_PUBKEY";
+
+        private static Regex validSignatureTagRegex = new Regex(_validTagSign, RegexOptions.Compiled);
+        private static Regex goodSignatureTagRegex = new Regex(_goodSignature, RegexOptions.Compiled);
+
+        /// <summary>
+        /// Obtain the number of tag on current git revision.
+        /// </summary>
+        /// <returns>Returns the number of annotated or signed tag on current revision.</returns>
+        public int NumberOfTag { get; private set; }
+
+        /// <summary>
+        /// Obtain the tag verification message for all the tag in current git revision 
+        /// </summary>
+        /// <returns>Full concatenated string coming from GPG analysis on all tags on current git revision.</returns>
+        public string TagVerifyMessage { get; private set; }
+
+        public GitGpgController(IGitModule module, GitRevision revision)
+        {
+            _module = module;
+            _revision = revision;
+
+            NumberOfTag = _revision.Refs.Count(x => x.IsTag && x.IsDereference);
+            TagVerifyMessage = "";
+        }
+
+        /// <summary>
+        /// Obtain the commit signature status on current revision.
+        /// </summary>
+        /// <returns>Enum value that indicate the gpg status for current git revision.</returns>
+        public CommitStatus CheckCommitSign()
+        {
+            CommitStatus _cmtStatus = CommitStatus.NoSignature;
```

Comment:
here and everywhere else - please do not prefix local variables with `_` (don't prefix them at all). 
this prefix is generally used for class fields (thanks to Resharper 😒). it makes it harder to reason about the code.

Reply:
ok, it's only a matters of style but it's ok.